mailpoet/lib/Doctrine/EventListeners/NewsletterListener.php

```diff
@@ -0,0 +1,50 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Doctrine\EventListeners;
+
+use MailPoet\Entities\NewsletterEntity;
+use MailPoet\Entities\WpPostEntity;
+use MailPoet\WP\Functions as WPFunctions;
+use MailPoetVendor\Doctrine\ORM\Event\LifecycleEventArgs;
+
+class NewsletterListener {
+  /** @var WPFunctions */
+  private $wp;
+
+  public function __construct(
+    WPFunctions $wp
+  ) {
+    $this->wp = $wp;
+  }
+
+  public function prePersist(): void {
+    // the tests fail without this method
+  }
+
+  public function preUpdate(LifecycleEventArgs $eventArgs): void {
+    $entity = $eventArgs->getEntity();
+    if (!$entity instanceof NewsletterEntity) {
+      return;
+    }
+
+    $unitOfWork = $eventArgs->getEntityManager()->getUnitOfWork();
+
+    /** @var array{status: array{0: string, 1: string}} $changeSet */
```

Comment:
FYI, I was digging a bit more about this line required to avoid an error from PHPStan, and I found [this line in our PHPStan baseline](https://github.com/mailpoet/mailpoet/blob/d4d4db7cc5e796a3dbb8cc4e988bbfa0393fc4ca/mailpoet/tasks/phpstan/phpstan-8-baseline.neon#L209-L211). 😬

Reply:
Hmm, interesting. It didn't occur to me to check the baseline file. What do you suggest? Should we keep this line here, add this file to the base line as well or something else?

Reply:
I would keep it as it is. If I'm not wrong, we have a ticket for removing baseline files.
I wanted to let you know only the result of my finding.